src/Engine/ProtoCore/Namespace/ElementResolver.cs

```diff
@@ -0,0 +1,68 @@
+﻿using ProtoCore.AST.AssociativeAST;
+using ProtoCore.DSASM;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace ProtoCore.Namespace
+{
+    /// <summary>
+    /// Responsible for resolving a partial class name to its fully resolved name
+    /// </summary>
+    public class ElementResolver
+    {
+        private Dictionary<string, KeyValuePair<string, string>> resolutionMap;
```

Comment:
Should be made `readonly` (by the way, this class shouldn't be part of this PR, maybe all we need is an empty class declaration. It sorta *pollutes* the PR now).